libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/ConfigurationBotFrameworkAuthentication.cs

```diff
@@ -97,5 +105,17 @@ public override BotFrameworkClient CreateBotFrameworkClient()
         {
             return _inner.CreateBotFrameworkClient();
         }
+
+        /// <inheritdoc />
+        public override async Task<StreamingConnection> CreateWebSocketConnectionAsync(HttpContext httpContext, ILogger logger)
+        {
+            if (_useLegacyStreamingConnection)
+            {
+                var socket = await httpContext.WebSockets.AcceptWebSocketAsync().ConfigureAwait(false);
+                return new LegacyStreamingConnection(socket, logger);
+            }
+
+            return await base.CreateWebSocketConnectionAsync(httpContext, logger).ConfigureAwait(false);
+        }
```

Comment:
Given @carlosscastro's [comment](https://github.com/microsoft/botbuilder-dotnet/pull/5883#discussion_r714174437), I'm questioning the need for this.

This is a tactical change, but we should rationalize this against the v5 plan.

Reply:
It's also affected the Connector layer per my earlier comment about the muddling of SDK layers.

Reply:
Since CloudAdapter is shipped with legacy streaming, this config setting will allow existing customers to fall back to the legacy streaming if they need.